volatility/framework/plugins/windows/netscan.py

```diff
@@ -0,0 +1,306 @@
+
+        vers_offset = ntkrnlmp.get_symbol("KdVersionBlock").address
+
+        vers = ntkrnlmp.object(object_type = "_DBGKD_GET_VERSION64",
+                               layer_name = virtual_layer_name,
+                               offset = vers_offset)
+
+        vollog.debug("Determined OS Major/Minor Version: {}.{}".format(vers.MajorVersion, vers.MinorVersion))
+
+        vers_minor_version = int(vers.MinorVersion)
+
+        # this is a hard-coded address in the Windows OS
+        if virtual_layer.bits_per_register == 32:
+            kuser_addr = 0xFFDF0000
+        else:
+            kuser_addr = 0xFFFFF78000000000
+
+        kuser = ntkrnlmp.object(object_type = "_KUSER_SHARED_DATA",
+                                layer_name = virtual_layer_name,
+                                offset = kuser_addr,
+                                absolute = True)
+
+        nt_major_version = str(kuser.NtMajorVersion)
+        nt_minor_version = str(kuser.NtMinorVersion)
+
+        # default to general class types, may be overwritten later.
+        class_types = network.class_types
+        if nt_major_version == "10":
+            if arch == "x64":
+                # win10 x64 has an additional class type we have to include.
+                class_types = network.win10_x64_class_types
+
+            if vers_minor_version < 14393:
+                # all win10 below 14393 have the same structs for our needs.
+                filename = "netscan-win10-{arch}".format(arch=arch)
+            elif vers_minor_version < 15063:
+                if arch == "x64":
+                    filename = "netscan-win10-x64"
+                else:
+                    # 14393 x86 is special.
+                    filename = "netscan-win10-14393-x86"
+            else:
+                # for now all newer windows versions share the same structs.
+                filename = "netscan-win10-15063-{arch}".format(arch=arch)
+
+        elif nt_major_version == "6":
+            # win between vista and 8.1
+            if nt_minor_version == "0":
+                # vista
+                # if vista sp 12 x64 then:
+                # filename = "netscan-vista-sp12-x64"
+                filename = "netscan-vista-{arch}".format(arch=arch)
+            elif nt_minor_version == "1":
+                # 7
+                filename = "netscan-win7-{arch}".format(arch=arch)
+            elif nt_minor_version == "2":
+                # 8
+                filename = "netscan-win8-{arch}".format(arch=arch)
+            elif nt_minor_version == "3":
+                # 8.1
+                filename = "netscan-win81-{arch}".format(arch=arch)
+        else:
+            # default to a fallback, but this *should* not happen.
+            filename = "netscan-win{vers}-{arch}".format(vers=major_version, arch=arch)
+
+        vollog.debug("Determined symbol filename: {}".format(filename))
+
+        return filename, class_types
+
+    def create_netscan_symbol_table(self) -> str:
```

Comment:
I'd somewhat prefer to see this and `determine_tcpip_version` as `@classmethod`, so that other plugins can call them  It would require parameterizing them anywhere that self is used (so I'd expect a `context`, a `config_path` and a `symbol_table_name` parameter for this one, looks like `determine_tcpip_version` would just need `context`, `layer_name`, `symbol_table_name` parameters, that also keeps all parameters except the context as simple strings).

Reply:
done!